yt_dlp/extractor/twitcasting.py

```diff
@@ -79,61 +88,116 @@ def _real_extract(self, url):
         title = (clean_html(get_element_by_id('movietitle', webpage))
                  or self._html_search_meta(['og:title', 'twitter:title'], webpage, fatal=True))
 
-        video_js_data = {}
-        m3u8_url = self._search_regex(
-            r'data-movie-url=(["\'])(?P<url>(?:(?!\1).)+)\1',
-            webpage, 'm3u8 url', group='url', default=None)
-        if not m3u8_url:
-            video_js_data = self._parse_json(self._search_regex(
-                r'data-movie-playlist=(["\'])(?P<url>(?:(?!\1).)+)',
-                webpage, 'movie playlist', group='url', default='[{}]'), video_id)
-            if isinstance(video_js_data, dict):
-                video_js_data = list(video_js_data.values())[0]
-            video_js_data = video_js_data[0]
-            m3u8_url = try_get(video_js_data, lambda x: x['source']['url'])
+        video_js_data = try_get(
+            webpage,
+            lambda x: self._parse_json(self._search_regex(
+                r"data-movie-playlist='([^']+?)'",
+                x, 'movie playlist', default=None), video_id)["2"], list)
```

Comment:
```suggestion
        video_js_data = try_get(
            webpage,
            lambda x: self._parse_json(self._search_regex(
                r'data-movie-playlist=\'([^\']+?)\'',
                x, 'movie playlist', default=None), video_id)['2'], list)
```
I prefer using `'` and just escaping the other single quotes present in the pattern.

Reply:
yeah it should